app/modules/enduser_setup.c

```diff
@@ -1109,7 +1233,7 @@ static void enduser_setup_dns_recv_callback(void *arg, char *recv_data, unsigned
 
 static void enduser_setup_free(void)
```

Comment:
I don't know if that is safe. It doesn't sound safe to me. Let's try to avoid that behaviour.